src/content/articles/objects-narrowed-before-usage.mdx

```diff
@@ -0,0 +1,60 @@
+---
+date: "June 1 2022"
+description: "yippee!..."
+---
+
+# Why Objects Aren't Narrowed Before Usage
+
+```ts twoslash
+const counts = {
+  one: 1,
+  two: 2,
+};
+
+const twoValue = counts.two;
+//    ^?
+```
+
+Shouldn't TypeScript know that `counts.two` is specifically the literal type `2` and not the general primitive type `number`?
+Can't it tell we haven't changed the object yet.
+
+It can't, but it won't.
+And for very good reason.
+
+<!--truncate-->
+
+## Change Tracking is Impractical
```

Comment:
I have a sinking feeling that I'm missing some major point in this writeup. Either a major reason why TypeScript won't, or some reasoning within this one... will sleep on it before writing more.